lib/project_types/script/layers/infrastructure/assemblyscript_dependency_manager.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module Script
+  module Layers
+    module Infrastructure
+      class AssemblyScriptDependencyManager
+        def initialize(ctx, language, extension_point, script_name)
+          @ctx = ctx
+          @language = language
+          @script_name = script_name
+          @extension_point = extension_point
+        end
+
+        def bootstrap
+          write_npmrc
+          write_package_json
+        end
+
+        def installed?
+          # Assuming if node_modules folder exist at root of script folder, all deps are installed
+          Dir.exist?("node_modules")
+        end
+
+        def install
+          output, status = @ctx.capture2e("npm", "install", "--no-audit", "--no-optional", "--loglevel error")
+          raise DependencyInstallError, output unless status.success?
+        end
+
+        private
+
+        def write_npmrc
+          npmrc = <<~HERE
+            @shopify:registry=https://registry.npmjs.org/
+          HERE
+
+          File.write(".npmrc", npmrc)
+        end
+
+        def write_package_json
+          package_json = <<~HERE
+            {
+              "name": "#{@script_name}",
+              "version": "1.0.0",
+              "devDependencies": {
+                "@shopify/scripts-sdk-as": "#{@extension_point.sdks[:ts].sdk_version}",
+                "#{@extension_point.sdks[:ts].package}": "#{@extension_point.sdks[:ts].version}",
+                "@as-pect/assembly": "3.1.1",
+                "@as-pect/cli": "3.1.1",
+                "@as-pect/core": "3.1.1",
+                "yargs": "10.0.0",
+                "as-wasi": "^0.0.1",
+                "assemblyscript": "^0.9.4",
+                "ts-node": "^8.5.4",
+                "typescript": "^3.7.3"
+              },
+              "scripts": {
+                "test": "asp --config test/as-pect.config.js --summary --verbose"
+              }
+            }
+          HERE
+
```

Comment:
In the Node project-type in the CLI, we use `git` to `clone` repo that contains a prepared Node/React app skeleton (which includes a `package.json`), then call `npm install` to install the required packages.  Not sure if a similar approach could be taken here (cloning a repo to provide a scripts project skeleton).

Reply:
Cool, thanks for pointing that out! Thats something we're looking at now. We've extracted some stuff out into packages already but we're looking to make that better